stdlib/public/core/SequenceAlgorithms.swift

```diff
@@ -707,15 +707,29 @@ extension Sequence {
   /// - Complexity: O(*n*), where *n* is the length of the sequence.
   @inlinable
   public __consuming func reversed() -> [Element] {
-    // FIXME(performance): optimize to 1 pass?  But Array(self) can be
-    // optimized to a memcpy() sometimes.  Those cases are usually collections,
-    // though.
-    var result = Array(self)
-    let count = result.count
-    for i in 0..<count/2 {
-      result.swapAt(i, count - ((i + 1) as Int))
+    let underestimatedCount = self.underestimatedCount
+    if underestimatedCount == 0 {
+      var result = Array(self)
+      let count = result.count
+      for i in 0..<count/2 {
+        result.swapAt(i, count - ((i + 1) as Int))      }
+      return result
+    } else {
+      var iterator = makeIterator()
+      var result = [Element](unsafeUninitializedCapacity: underestimatedCount) { buf, initedCount in
```

Comment:
Nit: I think we're still respecting the 80-column formatting rule where possible. Also, "inited" is not an idiomatic English word, nor is it easily pronounceable in English; prefer "initialized."

Reply:
Fixed.